### PR TITLE
Update pipelines.md

### PR DESCRIPTION
**Pull request by stephanieatte**
Some provider settings where missing from the docs as per https://registry.terraform.io/providers/buildkite/buildkite/latest/docs/resources/pipeline#provider-settings-configuration

I added in the Properties available for Bitbucket Cloud, GitHub, and GitHub Enterprise: optional properties
build_branches                              build_pull_request_labels_changed  
cancel_deleted_branch_builds    
       
Additional properties available for GitHub:
 build_pull_request_ready_for_review

### CODE REVIEW

File: pages/apis/rest_api/pipelines.md [hunk 1]
```diff
@@ -1103,6 +1103,13 @@ Properties available for Bitbucket Cloud, GitHub, and GitHub Enterprise:
       <p class="Docs__api-param-eg"><em>Values:</em> <code>true</code>, <code>false</code></p>
     </td>
   </tr>
+  <tr>
+    <th><code>build_branches</code></th>
+    <td>
+      Whether to create builds when branches are pushed
```
Comment (dannymidnight) [inline pages/apis/rest_api/pipelines.md:1109]:
```suggestion
      Whether to create builds when branches are pushed.
```

File: pages/apis/rest_api/pipelines.md [hunk 2]
```diff
@@ -1146,6 +1153,12 @@ Properties available for Bitbucket Cloud, GitHub, and GitHub Enterprise:
       Whether to create a separate status for each job in a build, allowing you to see the status of each job directly in Bitbucket or GitHub.
       <p class="Docs__api-param-eg"><em>Values:</em> <code>true</code>, <code>false</code></p></td>
   </tr>
+  <tr>
+    <th><code>cancel_deleted_branch_builds</code></th>
+    <td>
+      A boolean to enable automatically cancelling any running builds for a branch if the branch is deleted
```
Comment (dannymidnight) [inline pages/apis/rest_api/pipelines.md:1159]:
```suggestion
      A boolean to enable automatically cancelling any running builds for a branch if the branch is deleted.
```

File: pages/apis/rest_api/pipelines.md [hunk 3]
```diff
@@ -1171,6 +1184,18 @@ Additional properties available for GitHub:
         Whether to create builds for pull requests from third-party forks.
         <p class="Docs__api-param-eg"><em>Values:</em> <code>true</code>, <code>false</code></p></td>
     </tr>
+    <tr>
+      <th><code>build_pull_request_labels_changed</code></th>
+      <td>
+        Whether to create builds for pull requests when labels are added or removed
```
Comment (dannymidnight) [inline pages/apis/rest_api/pipelines.md:1190]:
```suggestion
        Whether to create builds for pull requests when labels are added or removed.
```

**Review by dannymidnight — CHANGES_REQUESTED** [review]
Looks good, just some minor suggestions for consistent punctuation. 

(not blocking) This PR makes me wonder if we should think about deriving out rest API docs from the terraform provider schema 🤔

**Review by gilesgas — APPROVED** [review]

**Pull request merged**